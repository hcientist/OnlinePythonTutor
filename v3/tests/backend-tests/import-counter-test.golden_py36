{
  "code": "from collections import Counter\nc = Counter()\nc['foo'] += 1\nc['foo'] += 1\nc['foo'] += 1\nc['bar'] += 10\nc['baz'] = 100\n",
  "trace": [
    {
      "line": 1,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ]
      },
      "ordered_globals": [
        "Counter"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 3,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ],
        "c": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "Counter",
        "c"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE_PPRINT",
          "Counter",
          "Counter()"
        ]
      },
      "stdout": ""
    },
    {
      "line": 4,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ],
        "c": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "Counter",
        "c"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE_PPRINT",
          "Counter",
          "Counter({'foo': 1})"
        ]
      },
      "stdout": ""
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ],
        "c": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "Counter",
        "c"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE_PPRINT",
          "Counter",
          "Counter({'foo': 2})"
        ]
      },
      "stdout": ""
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ],
        "c": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "Counter",
        "c"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE_PPRINT",
          "Counter",
          "Counter({'foo': 3})"
        ]
      },
      "stdout": ""
    },
    {
      "line": 7,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ],
        "c": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "Counter",
        "c"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE_PPRINT",
          "Counter",
          "Counter({'bar': 10, 'foo': 3})"
        ]
      },
      "stdout": ""
    },
    {
      "line": 7,
      "event": "return",
      "func_name": "<module>",
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE",
          "imported class"
        ],
        "c": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "Counter",
        "c"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "INSTANCE_PPRINT",
          "Counter",
          "Counter({'baz': 100, 'bar': 10, 'foo': 3})"
        ]
      },
      "stdout": ""
    }
  ]
}
