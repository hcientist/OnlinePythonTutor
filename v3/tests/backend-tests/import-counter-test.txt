from collections import Counter
c = Counter()
c['foo'] += 1
c['foo'] += 1
c['foo'] += 1
c['bar'] += 10
c['baz'] = 100
