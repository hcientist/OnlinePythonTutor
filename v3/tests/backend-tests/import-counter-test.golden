{
  "code": "from collections import Counter\nc = Counter()\nc['foo'] += 1\nc['foo'] += 1\nc['foo'] += 1\nc['bar'] += 10\nc['baz'] = 100\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter", 
        "c"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "c": [
          "REF", 
          1
        ], 
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE_PPRINT", 
          "Counter", 
          "Counter()"
        ]
      }, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter", 
        "c"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "c": [
          "REF", 
          1
        ], 
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE_PPRINT", 
          "Counter", 
          "Counter({'foo': 1})"
        ]
      }, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter", 
        "c"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "c": [
          "REF", 
          1
        ], 
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE_PPRINT", 
          "Counter", 
          "Counter({'foo': 2})"
        ]
      }, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter", 
        "c"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "c": [
          "REF", 
          1
        ], 
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE_PPRINT", 
          "Counter", 
          "Counter({'foo': 3})"
        ]
      }, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter", 
        "c"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "c": [
          "REF", 
          1
        ], 
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE_PPRINT", 
          "Counter", 
          "Counter({'bar': 10, 'foo': 3})"
        ]
      }, 
      "line": 7, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "Counter", 
        "c"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "c": [
          "REF", 
          1
        ], 
        "Counter": [
          "IMPORTED_FAUX_PRIMITIVE", 
          "imported class"
        ]
      }, 
      "heap": {
        "1": [
          "INSTANCE_PPRINT", 
          "Counter", 
          "Counter({'baz': 100, 'bar': 10, 'foo': 3})"
        ]
      }, 
      "line": 7, 
      "event": "return"
    }
  ]
}
